### DISCUSSION TITLE
How to implement google global site tags?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#36983] [Help]
Below is an example of a google global site tag:

```
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=GA_MEASUREMENT_ID"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){window.dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'GA_MEASUREMENT_ID');
</script>
```
^^
Google indicates to add this tag on the head of the document.

I am using it in the _app.js file.

I am using the Head tag from Nextjs and it is rejecting the tag...  with a regular link tag I get a syntax error, but this is pretty much just a copy and paste from Google, there isn't much to change syntax.

If I use the Script tag from Next it tells me I shouldn't put it in the Head...  and still give me the same syntax error if I put it right under the head tag within the same div of the Head.

![error](https://user-images.githubusercontent.com/93617433/168835878-fd60e3b2-5059-43ba-b3f0-48a422a9fc9b.PNG)

**Comment by the-mgi** [comment]
this is how i did it, placed it in the _docuemnt.js, but you can place it in the _app.js as well. hope it helps.
![image](https://user-images.githubusercontent.com/70032816/168846199-642eff3f-f43b-4778-a9a0-eceac8804a81.png)

**Reply by michalvavra** [reply]
Use [`next/script`](https://nextjs.org/docs/basic-features/script) and do not put `<Script />` inside of `<Head />`.

You can look into official [with-google-analytics](https://github.com/vercel/next.js/tree/af86ca08e4363f2bbc10e4bd94a76b4f62b106c2/examples/with-google-analytics) example. Specifically [`pages/_app.js`](https://github.com/vercel/next.js/blob/af86ca08e4363f2bbc10e4bd94a76b4f62b106c2/examples/with-google-analytics/pages/_app.js).

```jsx
const App = ({ Component, pageProps }) => {
  return (
    <>
      <Script
        strategy="afterInteractive"
        src="https://www.googletagmanager.com/gtag/js?id=GA_MEASUREMENT_ID"
      />
      <Script
        id="gtag-init"
        strategy="afterInteractive"
        dangerouslySetInnerHTML={{
          __html: `
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
            gtag('config', 'GA_MEASUREMENT_ID');
          `,
        }}
      />
      <Component {...pageProps} />
    </>
  )
}

export default App
```

**Reply by Elindo586** [reply]
Thanks.... it isn't giving me an error... I will have to tests it later on...
